addons/edi/models/edi_document.py

```diff
@@ -687,6 +695,29 @@ def unlink(self):
             if edi_progress:
                 edi_progress.unlink()
 
+    def check_sequential_exec_allowed(self):
+        """If document type has sequential execution set, ensure previous documents of same type are
+        already processed (done/cancel)."""
+        if self.doc_type_id.execute_sequentially:
+            previous_docs = self.search_read(
+                [
+                    ("doc_type_id", "in", self.doc_type_id.ids), # Same document type
+                    ("state", "in", ("draft", "prep")),
+                    ("id", "<", self.id), # Previous documents _should_ have earlier ids
+                ],
+                ["name"],
+            )
+            if previous_docs:
+                self.message_post(
```

Comment:
Question: Do we want to create an issue (Project issue) or only  logging the message?